ivy/utils/_importlib.py

```diff
@@ -1,11 +1,95 @@
+import os
 import sys
+from typing import List, Set
+from pathlib import Path
 from importlib.util import resolve_name, module_from_spec
 from ivy.utils.backend import ast_helpers
 
 
 import_cache = {}
 path_hooks = []
 
+ivy_path_abs = Path(sys.modules["ivy"].__file__).parents[1]
+
+# Note that any modules listed as 'to skip' should not depend on the Ivy backend state.
+# If they do, the behavior of ivy.with_backend is undefined and may not function as
+# Expected.
+MODULES_TO_SKIP = ["ivy.compiler"]
+MODULES_TO_EXCLUDE = []
```

Comment:
I think this could have a clearer name, something like `MODULES_TO_KEEP/MODULES_TO_NOT_SKIP/MODULES_TO_IMPORT` since `MODULES_TO_EXCLUDE` has a very similar meaning to `MODULES_TO_SKIP`, saying that, what exactly is the purpose of `MODULES_TO_EXCLUDE`? I think it could be used to include specific modules within packages that are in `MODULES_TO_SKIP`, but is that valid?  (to have a global package but a local module from that package e.g. `MODULES_TO_SKIP = ['ivy.compiler'], MODULES_TO_EXCLUDE = ["ivy.compiler.utils"]`)

Reply:
It partially made sense to me back then, but you're right! this will actually have invalid references all around, I'll get the logic for MODULES_TO_EXCLUDE removed, great catch @Infrared1029 !